### PR TITLE
[DNM] priority_threshold

### PR DESCRIPTION
**Pull request by lawrencejones**
Configure Que workers to parameterise the job acquisition query by
priority threshold, enabling gradual transition from one pool of workers
to another.

This is **never intended to be merged**. We'll rely on this feature branch for the duration of the switchover, but it shouldn't make its way to master.

### CODE REVIEW

File: lib/que/sql.rb [hunk 1]
```diff
@@ -50,9 +50,10 @@ module Que
         FROM (
           SELECT j
           FROM que_jobs AS j
-          WHERE queue = $1::text
+          WHERE (queue = $1::text OR queue = '')
```
Comment (nlopes) [inline lib/que/sql.rb]: Just for my own understanding, why this change?

Comment (lawrencejones) [reply lib/que/sql.rb]: We want to run two sets of workers, one consuming jobs from a `gcp` queue and another from `softlayer`. We also want both of these queues to consume work from the default queue, as we won't be changing all jobs to be one queue or the other, we'll be using priority threshold to gradually permit GCP workers to take more jobs.

This should be documented [here](https://paper.dropbox.com/doc/Que-Migration-Dependencies-per-env-bBt1eYG1j3OZVhMIQF64U#:uid=061592766615478911359257&h2=Configuration)

Comment (nlopes) [reply lib/que/sql.rb]: Thanks for linking the doc, I had missed it!

**Comment by lawrencejones** [comment]
Going to close this PR, but leave the branch around. Don't want someone merging it by mistake. Will re-open if it requires additional changes.

**Comment by dmagliola** [comment]
👍 LGTM!

**Comment by lawrencejones** [comment]
Closing in favour of https://github.com/gocardless/que/pull/16 which is this work, but on top of the refactored master branch.
